packages/blocks/src/api/registration.js

```diff
@@ -758,3 +758,178 @@ export const registerBlockVariation = ( blockName, variation ) => {
 export const unregisterBlockVariation = ( blockName, variationName ) => {
 	dispatch( blocksStore ).removeBlockVariations( blockName, variationName );
 };
+
+/**
+ * Registers a new block bindings source with an object defining its
+ * behavior. Once registered, the source is available to be connected
+ * to the supported block attributes.
+ *
+ * @param {Object}   source                    Properties of the source to be registered.
+ * @param {string}   source.name               The unique and machine-readable name.
+ * @param {string}   source.label              Human-readable label.
```

Comment:
Label should be optional as it can be provided from the server.

Reply:
I made it optional in the pull request to bootstrap the source from the server: [link](https://github.com/WordPress/gutenberg/pull/63470/files#diff-30ce126b90014d659302087765876898f16562219f796796ed407fdb346c8e4dR769). Until then, I think label should be required, right?

Reply:
That works if you have a follow-up lined up.

Reply:
Yes, there is a follow-up already working. I just need to rebase and it should be ready.